executor/projection.go

```diff
@@ -85,6 +86,11 @@ func (e *ProjectionExec) Open(ctx context.Context) error {
 	if err := e.baseExecutor.Open(ctx); err != nil {
 		return err
 	}
+	failpoint.Inject("mockProjectionExecBaseExecutorOpenReturnedError", func(val failpoint.Value) {
+		if val.(bool) {
+			failpoint.Return(errors.New("mock e.baseExecutor.Open returned error"))
+		}
+	})
```

Comment:
We can place it in `baseExecutor.Open()` so that all tests can use the same failpoint.

Reply:
No, seems we can't, because if we inject error at baseExecutor, then all executor won't open

Reply:
and we can't control the granuity.